atlasdb-config/src/main/java/com/palantir/atlasdb/factory/AtlasDbServiceDiscovery.java

```diff
@@ -44,6 +45,15 @@ public static AtlasDbFactory createAtlasFactoryOfCorrectType(KeyValueServiceConf
                 TransactionKeyValueServiceManagerFactory.class);
     }
 
+    // TODO (jkong): A little cheaty, currently coupling usage of TKVSMF and KVSRF to be special or non-special...
```

Comment:
I'm not sure how cheaty this actually is. We're basically saying that any user who provides a custom TKVS must also provide a custom KVSR of the same type. I believe our current intention is to do exactly this.

Reply:
Yeah this is in line with our current plans for the classes. It still does feel a bit weird, since I'm only using the type from the config / if there was some configuration in the future for a KVSR _only_ (I don't know what this would be offhand, to be fair) , that shouldn't be a part of the TKVS config.